docs/userguide.md

```diff
@@ -0,0 +1,319 @@
+may be helpful to share some commonly used scripts so that not everyone will
+have to reimplement the wheel. Our download scripts for ERA5 (model and pressure
+level) data are shipped with the repository and can be found in the [scripts
+folder](https://github.com/WAM2layers/WAM2layers/tree/master/scripts). These
+scripts are currently not considered core functionality of WAM2layers and are
+therefore not part of the command line interface. Please see the docstrings
+inside these scripts for instructions on how to use them.
+
+```{note}
+It would be nice to collect more data download scripts over time. If you want to
+share your datascript, we will be happy to help you in adding it to our
+collection. Please see the [contributing](./contributing.md) chapter of this
+documentation.
+```
+
+### File structure
+
+A good file naming system (a.k.a. *data reference syntax*) makes it easy to
+process data for different date ranges, and to automatically read data inside
+our preprocessing scripts. In our download scripts we are using the following
+filenaming structure:
+
+```
+path/to/data/<year>/<month>/ERA5_<year>_<month>_<day><levtype>_<variable>.nc
+
+# for example:
+path/to/data/2021/07/ERA5_2021_07_15_ml_q.nc
+```
+
+Here, the `<levtype>` parameter can be `_ml` for model levels, `_pl` for
+pressure levels, and empty for surface variables. The `variable` name
+corresponds to the name of the variable inside the netcdf file, so if our file
+is called `......tp.nc` we expect to find a variable called `tp` inside this
+file.
+
+(pre-processing)=
+## Pre-processing
+
+During the pre-processing step we make sure to convert the raw data into
+something that the tracking script understands. This typically involves
+accumulating the data into two layers, deriving moisture fluxes (which are
+typically not included in standard model output) from variables that are
+commonly available, and converting the data to the right units and a regular
+grid if that isn't the case already.
+
+### Built-in preprocessing functionality
+
+WAM2layers comes with built-in preprocessing functionality for ERA5 data. You can
+use the following command:
+
+```
+# Preprocess era5 data using the wam2layers builting functionality
+wam2layers preprocess era5 config_file.yaml
+```
+
+where `config_file.yaml` is the path to your configuration file. This file
+should have settings on the date range for which you want to run the
+preprocessing, and also about the location where the raw data are stored and
+what filename pattern they follow. [Documentation on this](./config) is
+underway, but for now you can have a look at the example config file
+[here](https://github.com/WAM2layers/WAM2layers/blob/main/cases/era5_2021.yaml).
+
+```{note}
+For now WAM2layers only contains preprocessing code for era5. We think it would
+be nice to add preprocessing functionality for more datasets over time.
+```
+
+### Preprocessing other datasets
```

Comment:
under this header add note how to access/where to find the files that serve as example for a new preprocessing script

Reply:
TODO as part of #335